File: plugins/chrome-bootstrap/src/android/ChromeBootstrap.java
// Copyright (c) 2013 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package com.google.cordova;

import java.io.IOException;
import java.io.InputStream;
import java.io.ByteArrayInputStream;
import java.util.NavigableSet;
import java.util.TreeMap;

import org.apache.cordova.api.CordovaPlugin;
import android.util.Log;

public class ChromeBootstrap extends CordovaPlugin {

    private static final String LOG_TAG = "ChromeBootstrap";

    @Override
    public boolean execute(String action, CordovaArgs args, final CallbackContext callbackContext) throws JSONException {
        if ("doesNeedLoad".equals(action)) {
            callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK, true));
            return true;
        }
        return false;
    }

}

File: plugins/chrome-bootstrap/src/ios/ChromeBootstrap.m
// Copyright (c) 2013 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#import "ChromeBootstrap.h"

#pragma mark ChromeBootstrap

@implementation ChromeBootstrap

@synthesize needsLaunch=_needsLaunch;

- (CDVPlugin*)initWithWebView:(UIWebView*)theWebView
{
    self = [super initWithWebView:theWebView];
    if (self) {
        self.needsLaunch = ([[UIApplication sharedApplication] applicationState] != UIApplicationStateBackground);
    }
    return self;
}

- (void)doesNeedLaunch:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsBool:self.needsLaunch];
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

@end
